### PR TITLE
Add the migration warning

### PR DESCRIPTION
**Pull request by rememberlenny**
screen shot 2017-03-13 at 12 16 23 pm

### CODE REVIEW

**Review by rogeruiz — DISMISSED** [review]
I think we still have some changes to make to the copy or linking to the migration page before we can merge this into #39. But great work on getting this done so quickly @rememberlenny 🎉

**Comment by cnelson** [comment]
If you want the warning on the passcode page as well you are going to need to update the template resolver so that it knows to look in the environment specific paths for that template, otherwise the warning will appear in govcloud as well.  The change to the login template resolver was made here: https://github.com/18F/cg-uaa/commit/50e8bbf0cb

Although, I wonder if that really needs to be done, you don't get to see the passcode page without logging in first, so why the extra effort to display it twice?

**Comment by fureigh** [comment]
@cnelson Thanks for the pointers! It's important to include the passcode page because when a user logs in via the command line, they have to access the passcode page but they won't see the login page.

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (fureigh) [inline server/src/main/resources/templates/web/eastwest/login.html]: Let's link 'migrated your app(s)' to https://cloud.gov/docs/apps/govcloud/#migration. (Arguably, alerts don't usually have links, but adding this link will increase the chances that users are able to quickly answer their own questions instead of defaulting to emailing support.)

File: server/src/main/resources/templates/web/passcode.html
Comment (fureigh) [inline server/src/main/resources/templates/web/passcode.html]: Same suggestion as above: let's link 'migrated your app(s)' to https://cloud.gov/docs/apps/govcloud/#migration, if there are no objections.

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (fureigh) [inline server/src/main/resources/templates/web/eastwest/login.html]: It'd probably be kind to linkify the email addresses too. I don't feel strongly about it.

File: server/src/main/resources/templates/web/passcode.html
Comment (fureigh) [inline server/src/main/resources/templates/web/passcode.html]: Same suggestion as above: linkify email address.

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (fureigh) [inline server/src/main/resources/templates/web/eastwest/login.html]: Does the fixed-pixel `margin-bottom` hold up on mobile?

Comment (fureigh) [reply server/src/main/resources/templates/web/eastwest/login.html]: Also: is there an existing USWDS style that could be used instead?

**Comment by cnelson** [comment]
When a user logs in from the command line, they are forced to authenticate before seeing the passcode page, so they will see the login screen assuming it's been more than 15 minutes since last time they requested a passcode:

https://github.com/18F/cg-deploy-cf/blob/400e72f0ea9380806347f84f6671edcacf6feb5a/cf-properties.yml#L326

File: server/src/main/resources/templates/web/passcode.html
Comment (fureigh) [inline server/src/main/resources/templates/web/passcode.html]: Same questions about fixed-pixel specs and straight-out-of-the-USWDS alternatives.

File: server/src/main/resources/templates/web/passcode.html
Comment (fureigh) [inline server/src/main/resources/templates/web/passcode.html]: Same questions about fixed-pixel specs and straight-out-of-the-USWDS alternatives.

File: server/src/main/resources/templates/web/passcode.html
Comment (fureigh) [inline server/src/main/resources/templates/web/passcode.html]: Same questions about fixed-pixel specs and straight-out-of-the-USWDS alternatives.

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (fureigh) [inline server/src/main/resources/templates/web/eastwest/login.html]: Same questions about fixed-pixel specs and straight-out-of-the-USWDS alternatives.

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (fureigh) [inline server/src/main/resources/templates/web/eastwest/login.html]: Same questions about fixed-pixel specs and straight-out-of-the-USWDS alternatives.

File: server/src/main/resources/templates/web/passcode.html
Comment (fureigh) [inline server/src/main/resources/templates/web/passcode.html]: Same questions about fixed-pixel specs and straight-out-of-the-USWDS alternatives.

**Review by fureigh — CHANGES_REQUESTED** [review]
🎉 Thank you so much for putting this together! I've included some suggestions and questions (really, the same few suggestions/questions throughout). Ping me if you have any questions or concerns.

**Comment by fureigh** [comment]
@cnelson Ah, right you are! Having the alert appear only on the login screen should be fine, then. Thanks for the catch!

**Comment by rogeruiz** [comment]
This won't be making into the deployment for the UAA updates to 18F/cg-cf-release#37 so take your time. No longer a fire 🔥 👍

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (jcscottiii) [reply server/src/main/resources/templates/web/eastwest/login.html]: +1 on this one

File: server/src/main/resources/templates/web/passcode.html
Comment (jcscottiii) [reply server/src/main/resources/templates/web/passcode.html]: +1 on this one

File: server/src/main/resources/templates/web/passcode.html
Comment (fureigh) [inline server/src/main/resources/templates/web/passcode.html]: @rememberlenny This file's changes can be backed out now, per @cnelson's observation that even CLI users have to go through the login screen too before they can reach this page.

**Comment by rememberlenny** [comment]
@fureigh @rogeruiz Updated design for single page change, and use of USWDS styles.

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (fureigh) [inline server/src/main/resources/templates/web/eastwest/login.html]: I suggest indenting this whole block for readability (specifically, so the nesting is easier to parse).

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (fureigh) [inline server/src/main/resources/templates/web/eastwest/login.html:51]: Would the 'dark background' styling from https://standards.usa.gov/components/typography/#links work instead of inline styling?

File: server/src/main/resources/templates/web/eastwest/login.html
Comment (fureigh) [inline server/src/main/resources/templates/web/eastwest/login.html:54]: Same question as line 51.

**Review by fureigh — CHANGES_REQUESTED** [review]
@rememberlenny Thanks for these changes! Back to you with a couple of suggested tweaks.

ping-pong-pigeons

**Comment by fureigh** [comment]
ping-pong-pigeons

**Comment by fureigh** [comment]
@rememberlenny If you're swamped today but :cool: with those changes, ping me and I can make 'em. Or merge as-is and we can optimize later; it's only for a couple of days.

**Review by fureigh — APPROVED** [review]

**Comment by fureigh** [comment]
LGTM. Thanks, @rememberlenny!

**Pull request merged**
